### ISSUE TITLE
Timer test fails

### ISSUE THREAD

**Issue by alfred-bratterud** [hioa-cs/IncludeOS#545]
After the new APIC / MSI-x pr #543  was merged `./test/timers` fails just when the last timer is supposed to trigger, due to a CPU exception:

```
1s. PULSE #25 
                [x] After 25 sec, 1s timer x 30 == 25 times, 2s timer x 15 == 10 times
                [x] There are now 2 timers left

ISR: 0x0  IRR: 0x0

[0] Return 0x2091ff
[1] Return 0x26441e
[2] Return 0x2035bd
[3] Return 0x2021ec
[4] Return 0x202045
[5] Return 0x201ce1
[6] Return 0x464c457f
>>>> !!! CPU EXCEPTION !!! <<<<
Heap end: 0x2d3000 

```

**Comment by alfred-bratterud** [comment]
I'm getting the same result consistently on Virtualbox and bare metal KVM

**Comment by alfred-bratterud** [comment]
Setting the Qemu `-smp` parameter to 1 removes the problem as well.

**Comment by alfred-bratterud** [comment]
Verified on both bare metal KVM, VirtualBox and nested in openStack (Jenkins).

**Comment by alfred-bratterud** [comment]
Fixed in dev after #578. I'll consider hotfixing master with this. As this only applies to stopping repeated timers, which isn't used anywhere in the core system, it shouldn't cause instability unless your service stops a repeated timer.
